docs/settings.md

```diff
@@ -111,6 +111,21 @@ L0 delta layer threshold for L1 iamge layer creation. Default is 3.
 
 WAL retention duration for PITR branching. Default is 30 days.
 
+#### walreceiver_connect_timeout
+
+Time to wait to establish the wal receiver connection before failing
+
+#### lagging_wal_timeout
+
+Time the pageserver did not get any WAL updates from safekeeper (if any).
+Avoids lagging pageserver preemptively by forcing to switch it from stalled connections.
+
+#### max_lsn_wal_lag
+
+Difference between Lsn values of the latest available WAL on safekeepers: if currently connected safekeeper starts to lag too long and too much,
+it gets swapped to the different one.
+A safekeeper has to be both lagging beyond the threshold and do that over `lagging_wal_timeout` time in order to be considered as lagging.
```

Comment:
@arssher WDYT?

Reply:
I think these conditions should be ORed, and ANDed -- either of them is trigger to switch to more advanced safekeeper.
The idea is, relying only on timeout is not enough because safekeeper might lag a lot and slowly catch up. Relying only on lag is not enough because we might be in deadlock when new WAL is not generated because old hasn't reached pageserver. However, both values must be big enough to prevent flip-flopping during normal work.

Reply:
... and looking at the code, these are ORed actually, so just doc needs a fix.

Reply:
Actually, the code has both conditions to check also, but apparently it's written a bit complicated, since fooled a second person already.

I've got your point and will apply OR everywhere.